test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/ProjectRestoreInfoBuilder.cs

```diff
@@ -90,12 +106,22 @@ public ProjectRestoreInfoBuilder WithTool(string name, string version)
         public ProjectRestoreInfoBuilder WithTargetFrameworkInfo(
             IVsTargetFrameworkInfo tfi)
         {
-            (_pri.TargetFrameworks as VsTargetFrameworks).Add(tfi);
+            if(_projectRestoreInfo.TargetFrameworks is VsTargetFrameworks && tfi is VsTargetFrameworkInfo)
```

Comment:
Cast with pattern matching to avoid extra casts afterwards.

    if (_projectRestoreInfo.TargetFrameworks is VsTargetFrameworks vsTargetFrameworks && tfi is VsTargetFrameworkInfo)